content/articles/creating-an-animated-javafx-field-validation-alert/index.md

```diff
@@ -0,0 +1,758 @@
+
+![When the login is successful](/engineering-education/creating-an-animated-javafx-field-validation-alert/on-success-message.png "When the login is successful")
+
+### Add animations to the project
+
+To add animations in the JavaFX project, the `AnimateFX` dependency is required.
+
+> `AnimateFX` is a JavaFX library that provides animations for a JavaFX project.
+
+AnimateFX animations for JavaFX controls include:
+
+- Bounce
+- BounceIn
+- FadeIn
+- Flash
+- Flip
+- GlowText
+- Jello
+- LightSpeedIn
+- Pulse
+- RollIn
+- RotateOut
+- RubberBand
+- SlideOutRight
+- Swing
+- Wobble
+- ZoomOut
+
+Check out for additional animations [here](https://github.com/Typhon0/AnimateFX/tree/master/animatefx/src/main/java/animatefx/animation).
+
+- Check the dependencies installed or used in the project by clicking on the `Dependencies` button near the status bar.
+It opens up the Dependencies tab.
+Search for `AnimateFX` and add it to the project.
+When searched, it looks as shown below:
+
+![AnimateFX dependency](/engineering-education/creating-an-animated-javafx-field-validation-alert/animateFX-dependency.png "animateFX-dependency")
+
+- Wait for a few minutes for the download to complete.
+Once done, reload the file or IDE by closing and re-opening it.
+Check if it has been installed in the same Dependencies tab.
+
+- Once installed head over to the `HelloController.java` file.
+- Inside the `onLoginButtonClick()` function, find the procedure checking if the username and password are blank. In it add the code below:
+
+```java
+                new animatefx.animation.Shake(usernameTextField).play();
+                new animatefx.animation.Wobble(usersIcon).play();
+                new animatefx.animation.Shake(userPassword).play();
+                new animatefx.animation.Wobble(passwordIcon).play();
+```
+
+This code will add a '**Shake**' animation to the inputs when there is an error. It also adds a '**Wobble**' animation to the images.
+
+- In the procedure that checks if the username is blank, add the following code to it:
+
+```java
+                    new animatefx.animation.Shake(usernameTextField).play();
+                    new animatefx.animation.Pulse(usersIcon).play();
+```
+
+The code adds the '**Pulse**' and '**Shake**' effect to the input and icon respectively.
+
+- In the procedure that checks if the username is blank, add the following code:
+
+```java
+                    new animatefx.animation.Shake(userPassword).play();
+                    new animatefx.animation.Wobble(passwordIcon).play();
+```
+
+The code also does as the previous one but now to the PasswordField and its icon.
+
+- In the procedure that checks if the password is less than four characters, add a slightly gentle effect.
+Add the following code to it:
+
+```java
+                new animatefx.animation.FadeIn(userPassword).play();
+                new animatefx.animation.Wobble(passwordIcon).play();
+```

```

Comment:
Format your code to start from the margin of the document. Do this for all your snippets.

Reply:
It was much...but all done!